scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -658,8 +640,22 @@ def apply_filters(self, **kwargs):
                     if not set(ts.modules).issubset(set(self.modules)):
                         instance.add_filter(f"one or more required modules not available: {','.join(ts.modules)}", Filters.TESTSUITE)
 
-                if runnable and not instance.run:
-                    instance.add_filter("Not runnable on device", Filters.PLATFORM)
+                # instance.run tells twister if a test can be build and executed (if true) or only build (if false)
+                instance.run, verdict = instance.check_runnable(self.options.enable_slow, runnable)
+
+                harness_supported, verdict_harness = instance.is_harness_supported()
+                if not harness_supported:
+                    instance.run = False
+                    verdict = verdict_harness
+
+                fixture_runnable, verdict_fixture = instance.is_fixture_available(plat.name, self.options.fixture, self.hwm.duts)
+                if not fixture_runnable:
+                    instance.run = False
+                    verdict = verdict_fixture
+
+                if runnable:
+                    if not instance.run:
+                        instance.add_filter(f"Not runnable on device: {verdict}", Filters.PLATFORM)
```

Comment:
@evgeniy-paltsev here? I used fixup and rebased, so no new commit. Looks like you suggested with extra `because` but I guess changing `.` to `:` keeps the gist of your comment?

Reply:
Ah, I see. Haven't notice the extra dot. :)
Thanks!